icu4c/source/common/uresdata.h

```diff
@@ -500,6 +500,9 @@ class ResourceDataValue : public ResourceValue {
     const ResourceData *pResData;
 
 private:
+    // TODO: If UResourceBundle.fResData becomes a pointer,
+    // then remove this field again.
+    ResourceData resData;
```

Comment:
I really don't like exposing more implementation details. I resolved the ownership issue and copyability by removing the pResData pointer from ResourceDataValue, and replacing it with a getter that returns a reference to the internal resData. When we refactor it to just store the resData once, then this will become a simple pointer again without changing much else.